packages/jaspr_lints/lib/src/assists/component_assist.dart

```diff
@@ -124,46 +58,130 @@ class ComponentAssistProvider extends DartAssist {
         }
       }
 
-      reporter.createChangeBuilder(priority: 1, message: 'Convert to StatefulComponent').addDartFileEdit((builder) {
-        builder.addReplacement(node.extendsClause!.superclass.sourceRange, (edit) {
-          edit.write('StatefulComponent');
-        });
+      convertToStatefulComponent(reporter, node, buildMethod);
+      if (buildMethod != null && buildMethod.body.keyword != null) {
+        convertToAsyncStatelessComponent(reporter, node, buildMethod);
+      }
+    });
+  }
 
-        var splitToken = buildMethod?.beginToken ?? node.rightBracket;
-        var indent = buildMethod != null ? '' : '  ';
-        var endIndent = buildMethod != null ? '  ' : '';
-        var name = node.name.lexeme;
+  void createStatelessComponent(
+      ChangeReporter reporter, SourceRange target, String nameSuggestion, bool hasJasprImport) {
+    reporter.createChangeBuilder(priority: 1, message: 'Create StatelessComponent').addDartFileEdit((builder) {
+      builder.addInsertion(target.end == 0 ? 1 : target.end, (edit) {
+        edit.write('class ');
+        edit.addSimpleLinkedEdit('name', nameSuggestion);
+        edit.write(' extends StatelessComponent {\n'
+            '  const ');
+
+        edit.addSimpleLinkedEdit('name', nameSuggestion);
+        edit.write('({super.key});\n\n  @override\n  Iterable<Component> build(BuildContext context) sync* {\n'
+            '    yield ');
+        edit.addSimpleLinkedEdit('child', "div([])");
+        edit.write(';\n  }\n}\n');
+      });
+      if (!hasJasprImport) {
+        builder.addSimpleInsertion(0, "import 'package:jaspr/jaspr.dart';\n");
+      }
+    });
+  }
 
-        builder.addInsertion(splitToken.offset, (edit) {
-          edit.write("$indent@override\n  State createState() => ${name}State();\n"
-              "}\n\nclass ${name}State extends State<$name> {\n$endIndent");
-        });
+  void createStatefulComponent(
+      ChangeReporter reporter, SourceRange target, String nameSuggestion, bool hasJasprImport) {
```

Comment:
When line is too long, it should be splited by this way:

```
void createStatefulComponent(
      ChangeReporter reporter, 
      SourceRange target, 
      String nameSuggestion, 
      bool hasJasprImport,
) {
   ...
}
```

Reply:
done